### PR TITLE
Laboratory work #1, Zelekson Daniel - 19FPL1

### PR DESCRIPTION
**Pull request by daniilzelekson**

### CODE REVIEW

**Comment by dmitry-uraev** [comment]
The deadline is not that far - consider pushing changes so that we can review them.

File: lab_1/main.py [hunk 1]
```diff
@@ -9,10 +9,19 @@ def tokenize(text: str) -> list:
...
     :param text: the initial text
     :return: a list of lowercased tokens without punctuation
-    e.g. text = 'The weather is sunny, the man is happy.'
+    E.G.. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
-    pass
+    new_text = ""
+    text = text.lower()
+    trash = set("""1234567890-=!@#$%^&*()_+,./<>?;:'"[{}]"'""")
```
Comment (dmitry-uraev) [inline lab_1/main.py]: it is not trash. These are characters to remove from the string

File: lab_1/main.py [hunk 2]
```diff
@@ -9,10 +9,19 @@ def tokenize(text: str) -> list:
     Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
     :param text: the initial text
     :return: a list of lowercased tokens without punctuation
-    e.g. text = 'The weather is sunny, the man is happy.'
+    E.G.. text = 'The weather is sunny, the man is happy.'
```
Comment (dmitry-uraev) [inline lab_1/main.py]: nice (no)

File: lab_1/main.py [hunk 3]
```diff
@@ -9,10 +9,19 @@ def tokenize(text: str) -> list:
...
     """
-    pass
+    new_text = ""
+    text = text.lower()
+    trash = set("""1234567890-=!@#$%^&*()_+,./<>?;:'"[{}]"'""")
+    for c in text:
+        if not c in trash:
+            new_text+=c
+    new_text = new_text.split()
+
```
Comment (dmitry-uraev) [inline lab_1/main.py]: there probably should be one more line written, don t you think so? (return)

File: lab_1/main.py [hunk 4]
```diff
@@ -25,7 +34,20 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+    if type(tokens) == dict:
```
Comment (dmitry-uraev) [inline lab_1/main.py]: why such specific checks? better write general one `if not isinstance(tokens, list): return []` and avoid so many checks below

File: lab_1/main.py [hunk 5]
```diff
@@ -36,7 +58,15 @@ def calculate_frequencies(tokens: list) -> dict:
     e.g. tokens = ['weather', 'sunny', 'man', 'happy']
     --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
     """
-    pass
+    d = {}
```
Comment (dmitry-uraev) [inline lab_1/main.py]: bad variable naming. This naming will not pass lint checks. Rewrite it

File: lab_1/main.py [hunk 6]
```diff
@@ -49,7 +79,11 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list:
     top_n = 1
     --> ['happy']
     """
-    pass
+    max_list = list(zip(d.values(),d.keys()))
+    max_list.sort()
+    max_token = list(max_list[-1])
+    (max_token).pop(0)
+
```
Comment (dmitry-uraev) [inline lab_1/main.py]: there probably should be one more line written, don t you think so? (return) (x2)

File: lab_1/main.py [hunk 7]
```diff
@@ -69,7 +103,11 @@ def get_concordance(tokens: list, word: str, left_context_size: int, right_conte
     right_context_size = 3
     --> [['man', 'is', 'happy', 'the', 'dog', 'is'], ['dog', 'is', 'happy', 'but', 'the', 'cat']]
     """
-    pass
+    concordance = []
```
Comment (dmitry-uraev) [inline lab_1/main.py]: consider adding checks for tests, otherwise they will fail

**Review by dmitry-uraev — COMMENTED** [review]
Overall comment:
1. Not bad start, but fix current issues and go over your functions again and implement correct checks for unit tests and lint. 

2. NB! Do not forget about the start.py file. You should also implement it. 
It seems to me, that you are implementing lab 1 for score 6, right? Correct me, if I am mistaken. 

3, Last but not least. lab_1 should be implemented by the 11:00 pm October 2th

File: lab_1/main.py (hunk 2 above)
Comment (daniilzelekson) [reply lab_1/main.py]: it was just a commit testing...

File: lab_1/main.py (hunk 1 above)
Comment (daniilzelekson) [reply lab_1/main.py]: should it be renamed or should I remake the whole function?

Comment (dmitry-uraev) [reply lab_1/main.py]: no, just rename

File: lab_1/main.py (hunk 2 above)
Comment (dmitry-uraev) [reply lab_1/main.py]: okay, rewrite

**Comment by dmitry-uraev** [comment]
Daniil, just fix tests and lint and get green Pull Request. i am receiving labs till 11:00 tomorrow. Ask if something urgent.

**Comment by dmitry-uraev** [comment]
It is not final version. Check your PR status - it is red.

File: lab_1/text.txt [hunk 8]
```diff
@@ -0,0 +1 @@
+Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```
Comment (dmitry-uraev) [inline lab_1/text.txt:1]: nice one

**Comment by daniilzelekson** [comment]
it's done!

**Comment by dmitry-uraev** [comment]
it is red, I need the green one

File: lab_1/main.py (hunk 7 above)
Comment (dmitry-uraev) [reply lab_1/main.py]: I need more ifs here, why just so few? (do not rewrite, I ll ask you on October 9th if your PR ll pass)

**Comment by dmitry-uraev** [comment]
Congratz, my latin friend

**Comment by dmitry-uraev** [comment]
done

**Pull request closed without merging**